### PR TITLE
Issue 880 testing mpi on travis

### PR DESCRIPTION
**Pull request by wds15**
#### Submission Checklist

- [X] Run unit tests: `./runTests.py test/unit`
- [ ] Run cpplint: `make cpplint`
- [X] Declare copyright holder and open-source license: see below

#### Summary:

This PR adds testing of MPI under linux using g++-4.9 and clang++-3.8 on Travis.

#### Intended Effect:
Improve test coverage of MPI. Note that only the folders which contain mpi tests are actually being tested with the Travis MPI environment.

For reviewer: If deemed desireable, it should be possible to get this to work as well under macOS. This is not as straightforward, but if this is seen valuable we can do so. Note that we concentrate primarily on Linux for the MPI feature at the moment, but macOS is claimed to be supported as well. With testing under macOS on Travis, we may consider to turn off MPI testing on Jenkins.

#### How to Verify:
Check the .travis.xml file which now has two more test environments and two more test cases.

#### Side Effects:
None.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Sebastian Weber

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,17 +30,39 @@ linux_gcc: &linux_gcc
...
   include:
     - <<: *osx_clang
       env:
         - MATRIX_EVAL="CXX=\"clang++ -DSTAN_THREADS\""
           TESTFOLDER="test/unit/math/rev/core test/unit/math/rev/mat/functor"      
-    - <<: *linux_clang
+    - <<: *mpi_linux_clang
       env:
-        - MATRIX_EVAL="CXX=clang++-3.8"
-          TESTFOLDER="test/unit/math/memory test/unit/math/prim/scal test/unit/math/prim/arr"
```
Comment (seantalts) [inline .travis.yml]: Did you mean to get rid of the memory, prim/scal, and prim/arr tests for linux_clang?

Comment (wds15) [reply .travis.yml]: Oh, NO! Let me double check this (this PR is to increase test-converage).

Comment (wds15) [reply .travis.yml]: Looks like I changed the lines towards what I wanted them without copying them first. Fixed now. Thanks for catching.

**Comment by bob-carpenter** [comment]
Good catch, Sean---this is why code review's so useful.  To pull out non-idiomatic code (hey, have you heard of lock_guard?) and to catch typo-driven mistakes (hey, did you drop those other tests on purpose?).

> Oh, NO! Let me double check this (this PR is to increase test-converage).

**Comment by wds15** [comment]
@seantalts all good now? I would feel better if this MPI stuff gets regularly tested in Travis as well.

**Comment by wds15** [comment]
thanks!

**Pull request merged**